### PR TITLE
add(print): support assuming a role after the initial account login role

### PR DESCRIPTION
**Pull request by cstavro**

### CODE REVIEW

File: saml/serviceProviders/aws/provider.go [hunk 1]
```diff
@@ -89,6 +90,37 @@ func (a AwsServiceProvider) GetCredentials(saml string, desiredRole string) *sts
...
+		})},
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	stsClient := sts.New(awsSession)
+	input := &sts.AssumeRoleInput{
+		RoleArn: aws.String(targetRole),
+		RoleSessionName: aws.String("cstavro-test"),
```
Comment (rtkjbeverly) [inline saml/serviceProviders/aws/provider.go:108]: This'll need to get swapped out to something less testy, but I don't think we need anything complicated to start with. The current model uses the [PID+username](https://github.com/rtkwlf/awn/blob/master/pkg/rtkokta/rtkokta/awsokta.py#L163)

File: print.go [hunk 2]
```diff
@@ -64,6 +65,11 @@ func Print(idProvider identityProviders.IdentityProvider, awsProvider servicePro
 	}
 
 	creds := awsProvider.GetCredentials(saml, printOptions.Role)
+
+	if printOptions.AssumeRole != "" {
+		creds, err = awsProvider.AssumeRole(*creds, printOptions.AssumeRole)
```
Comment (rtkjbeverly) [inline print.go:70]: I think this'll need a check on the `err`

**Review by rtkjbeverly — COMMENTED** [review]
Minor things, but LGMT.

For bringing in bmx, I'd like to get it working in our use case before worrying too much about being tidy.

**Comment by cstavro** [comment]
Pulling this out of draft because I use this functionality regularly and am getting tired of having to rebuild locally everytime I update my local env.

This works but I think there may be some confusion with the config property names. Happy to discuss the specifics of those.

File: print.go (hunk 2 above)
Comment (cstavro) [reply print.go:70]: Probably. Do we just panic or is there a more graceful way out of here that we want to use?

File: saml/serviceProviders/aws/provider.go (hunk 1 above)
Comment (cstavro) [reply saml/serviceProviders/aws/provider.go:108]: hmm, what username are you thinking here? There's no access to a username anywhere here. The default ~/.aws/config that we generate includes a 'role_session_name' that uses email address. Should we make this part of the .bmx config? Maybe default it to the USER env var? That seems to be my macos username. On windows it's USERNAME, should be easy enough to account for both.

File: print.go (hunk 2 above)
Comment (rtkjbeverly) [reply print.go:70]: For now I think a `log.Fatal(err)` is fine, as it happens above with authenticate.

File: saml/serviceProviders/aws/provider.go (hunk 1 above)
Comment (rtkjbeverly) [reply saml/serviceProviders/aws/provider.go]: I think just using the okta username would be sufficient, and we should already have it in the context from `printOptions.User` to be passed to AssumeRole as the 'roleSessionName' context.

**Comment by rtkjbeverly** [comment]
Resolved in #19 that is a subbranch of the work here, with modifications to resolve merge conflicts and pass in role session name.

**Pull request closed without merging**